lib/plaindate.ts

```diff
@@ -124,63 +122,63 @@ export class PlainDate implements Temporal.PlainDate {
     return ES.CreateTemporalDate(isoDate, calendar);
   }
   withCalendar(calendarParam: Params['withCalendar'][0]): Return['withCalendar'] {
-    if (!ES.IsTemporalDate(this)) throw new TypeErrorCtor('invalid receiver');
+    if (!ES.IsTemporalDate(this)) throw new TypeError('invalid receiver');
     const calendar = ES.ToTemporalCalendarIdentifier(calendarParam);
     return ES.CreateTemporalDate(GetSlot(this, ISO_DATE), calendar);
   }
   add(temporalDurationLike: Params['add'][0], options: Params['add'][1] = undefined): Return['add'] {
-    if (!ES.IsTemporalDate(this)) throw new TypeErrorCtor('invalid receiver');
+    if (!ES.IsTemporalDate(this)) throw new TypeError('invalid receiver');
     return ES.AddDurationToDate('add', this, temporalDurationLike, options);
   }
   subtract(
     temporalDurationLike: Params['subtract'][0],
     options: Params['subtract'][1] = undefined
   ): Return['subtract'] {
-    if (!ES.IsTemporalDate(this)) throw new TypeErrorCtor('invalid receiver');
+    if (!ES.IsTemporalDate(this)) throw new TypeError('invalid receiver');
     return ES.AddDurationToDate('subtract', this, temporalDurationLike, options);
   }
   until(other: Params['until'][0], options: Params['until'][1] = undefined): Return['until'] {
-    if (!ES.IsTemporalDate(this)) throw new TypeErrorCtor('invalid receiver');
+    if (!ES.IsTemporalDate(this)) throw new TypeError('invalid receiver');
     return ES.DifferenceTemporalPlainDate('until', this, other, options);
   }
   since(other: Params['since'][0], options: Params['since'][1] = undefined): Return['since'] {
-    if (!ES.IsTemporalDate(this)) throw new TypeErrorCtor('invalid receiver');
+    if (!ES.IsTemporalDate(this)) throw new TypeError('invalid receiver');
     return ES.DifferenceTemporalPlainDate('since', this, other, options);
   }
   equals(otherParam: Params['equals'][0]): Return['equals'] {
-    if (!ES.IsTemporalDate(this)) throw new TypeErrorCtor('invalid receiver');
+    if (!ES.IsTemporalDate(this)) throw new TypeError('invalid receiver');
     const other = ES.ToTemporalDate(otherParam);
     if (ES.CompareISODate(GetSlot(this, ISO_DATE), GetSlot(other, ISO_DATE)) !== 0) return false;
     return ES.CalendarEquals(GetSlot(this, CALENDAR), GetSlot(other, CALENDAR));
   }
   toString(options: Params['toString'][0] = undefined): string {
-    if (!ES.IsTemporalDate(this)) throw new TypeErrorCtor('invalid receiver');
+    if (!ES.IsTemporalDate(this)) throw new TypeError('invalid receiver');
```

Comment:
We duplicate this line 100+ times, with only the test varying. Should we have `ES.RejectReceiver(this, ES.IsTemporalDate)` ? 

This code is highly compressible so I'm not sure how much bundle size win that'd be, but it can't hurt and it'd probably make these files easier to read.

Reply:
Surprisingly large win: -4k minified.

Reply:
Cool. I suspect the diff in compressed & minified won't be nearly as large, but any improvement is good!